sympy/solvers/bivariate.py

```diff
@@ -196,6 +202,41 @@ def _solve_lambert(f, symbol, gens):
       a = -1, d = a*log(p), f = -log(d) - g*log(p)
     """
 
+    def _solve_even_degree_expr(expr, symbol):
+        """
+        Helper function to solve an expr containing dummy variable t at even
+        degree places to replace them with +symbol and -symbol.
+
+        For example:
+        Equation 2*log(t) + x/2 - log(a) = 0 will return solutions
+        of both of these equations.
+        1. 2*log(x) + x/2 - log(a) = 0
+        2. 2*log(-x) + x/2 - log(a) = 0
+
+        Parameters
+        ==========
+
+        expr : Expr
+            The expression which includes a dummy variable t which can be
+            replaced with +symbol and -symbol.
+
+        symbol : Symbol
+            The symbol concerned in expr.
+
+        Returns
+        =======
+
+        Solution of two equations generated by replacing t(at even degree places)
+        with positive and negative symbol.
+
+        """
+        plhs, nlhs = (expr.xreplace({t: s*symbol}) for s in (1, -1))
+        if plhs == nlhs:
+            return _solve_lambert(plhs, symbol, gens)
+        else:
+            psol, nsol = map(lambda f: _solve_lambert(f, symbol, gens), (plhs, nlhs))
```

Comment:
Here it is again :-)

```python
from sympy.utilities.iterables import uniq  <-- locate at top of file or function
sols = [_solve_lambert(f, symbol, gens) for f in (nlhs, plhs)]
return list(uniq(sols))
```
This will put the nlhs solution first -- think canonical -- and will will only return 1 if they are the same.

Reply:
> nlhs solution first -- think canonical 

Yeah I agree.
> Here it is again :-)

Here it is again from me :-). You forgot that the sols will be like this `[[sol of nlhs], [sol of plhs]]`. So I am using `return list(uniq(flatten(sols)))`.

Reply:
LOL -- let's see if you used a list comp or a map...